contracts/main/price-oracles/DexPriceOracle.sol

```diff
@@ -22,7 +23,9 @@ contract DexPriceOracle is Initializable, IFathomDEXOracle {
         returns (uint256, uint256)
     {
         require(token0 != token1, "DexPriceOracle/same-tokens");
-        (uint256 r0, uint256 r1) = FathomSwapLibrary.getReserves(dexFactory, token0, token1);
+        
+        address pair = FathomSwapLibrary.pairFor(dexFactory, token0, token1);
+        (uint256 r0, uint256 r1,) = IFathomSwapPair(pair).getReserves();
```

Comment:
2.2.7 There is no validation of return data in
FathomSwapLibrary

audit fix.

Instead of getReserves fn with staticCall, interface will be used in DexPriceOracle.sol and SlidingWindowDexOracle.sol